MCE-18519_reintroduce_go_mysql_error_handling error handling logic

smashleon23 opened the pull request:
replaying MCE-17245 that was left pending when upgrading with upstream

smashleon23 committed:
MCE-18519_reintroduce_go_mysql_error_handling error handling logic

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/instructure/mc-go-mysql?pullRequest=4) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/instructure/mc-go-mysql?pullRequest=4) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/instructure/mc-go-mysql?pullRequest=4) it.</sub>

adamcrown reviewed (approved):
Looks good to me. The linting is complaining about a few minor things. Some of which seem to be unrelated to these changes. I don't think  any of them need to be blocking though.

forain committed:
MCE-18894 Load Binlog Position from Redis

This was already implemented before the go-mysql MySQL 8 upgrade
but we missed refactoring this part after we merged upstream code.

Co-authored-by: Mario Poveda <mario.poveda@instructure.com>

smashleon23 committed:
MCE-18519_reintroduce_go_mysql_error_handling fix style warning
